client/checkout/api/index.js

```diff
@@ -320,12 +320,21 @@ export default class WCPayAPI {
 			action: 'create_payment_intent',
 			// eslint-disable-next-line camelcase
 			_ajax_nonce: getConfig( 'createPaymentIntentNonce' ),
-		} ).then( ( response ) => {
-			if ( ! response.success ) {
-				throw response.data.error;
-			}
-			return response.data;
-		} );
+		} )
+			.then( ( response ) => {
+				if ( ! response.success ) {
+					throw response.data.error;
+				}
+				return response.data;
+			} )
+			.catch( ( error ) => {
+				if ( error.message ) {
+					throw error;
+				} else {
+					// Covers the case of error on the Ajax request.
+					throw new Error( error.statusText );
```

Comment:
Lovin' it: this actually looks like it gets formatted pretty nicely inside `showError`, so happy days.